maup/repair.py

```diff
@@ -1,27 +1,1217 @@
 import math
 import pandas
+import geopandas
+from geopandas import GeoSeries, GeoDataFrame
+
 import functools
-import warnings
 
-from geopandas import GeoSeries, GeoDataFrame
-from shapely.geometry import MultiPolygon, Polygon
-from shapely.ops import unary_union
-# from shapely.validation import make_valid # currently in alpha
+import numpy
+
+import shapely
+from shapely import make_valid
+from shapely.ops import unary_union, split, polygonize, linemerge
+from shapely.geometry import (
+    Polygon,
+    MultiPolygon,
+    Point,
+    MultiPoint,
+    LineString,
+    MultiLineString,
+)
+from shapely.strtree import STRtree
 
 from .adjacencies import adjacencies
 from .assign import assign_to_max
 from .crs import require_same_crs
 from .indexed_geometries import get_geometries
 from .intersections import intersections
+from .progress_bar import progress
+
+from itertools import combinations
+from math import sqrt
+
+import warnings
+
+warnings.filterwarnings("ignore", "GeoSeries.isna", UserWarning)
+
+pandas.options.mode.chained_assignment = None
+
+
+"""
```

Comment:
See discussion about docstrings below—I found this comment to be super helpful, and IMO it would be even more helpful if parts of it were broken up and integrated inline with the implementation!